### PR TITLE
Split gh actions into smaller jobs

### PR DESCRIPTION
**Pull request by OchiengEd**
This pull request aims to split the Github actions workflow from a huge single job into smaller jobs.

### CODE REVIEW

**Comment by komish** [comment]
@OchiengEd The latest CI run failed because the job that installs Chart Verifier is not getting the correct value.
See: https://github.com/openshift-helm-charts/sandbox/actions/runs/6029387679/job/16358812490?pr=65041
You see the error is listing that the openshift-tools-installer did not get which binary it should be installing.

This is likely the cause for the blanket failure, so lets fix that and then we should be able to try again and see how we do.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -157,15 +173,26 @@ jobs:
...
+      message_file: ${{ steps.pr_comment.outputs.message-file }}
+
+    steps:
       - name: install chart verifier for action
         uses: redhat-actions/openshift-tools-installer@v1
         with:
           source: github
           skip_cache: true
-          chart-verifier: ${{ steps.set-env.outputs.verifier-action-image }}
+          chart-verifier: ${{ needs.setup-ci.outputs.verifier-action-image }}
```
[the remark this answers is not in the document]
Comment (OchiengEd) [reply .github/workflows/build.yml:192]: The syntax is `job.<job ID>.outputs.<output Name>`. Given the `set-env` task belongs to the `setup-ci` job. Therefore, to use it in steps in other jobs, we expose the output at the job level as shown below and then reference it via the job name as seen in line 192.

https://github.com/openshift-helm-charts/development/blob/5e9af25348771c1ad5d4fcb5e993101da6a9a9e7/.github/workflows/build.yml#L7-L14

If the `prepare-branch` job, which uses `${{ needs.setup-ci.outputs.run_build }}` completes, which it did, the syntax as a whole is correct. However, I do agree there is something amiss.

**Comment by komish** [comment]
Just a heads up that the current failure may be related to a new feature in chart-verifier which waits for statefulsets to complete before completing chart-testing. The vault chart has a statefulset that fails to become ready because of a readinessProbe checking readiness of the application. The vault chart, by default, is uninitialized and unsealed (because those actions are for the user), and as such the statefulset never becomes ready, which means that chart-testing doesn't run, leading to the timeout.

We can resolve this by modifying the values.yaml in our fixtures, but will also need to have Hashicorp resolve this in their certified chart down the road.

I'll look into options and get back to you once the fixtures have been updated.

**Comment by OchiengEd** [comment]
This was reworked in #265

**Pull request closed without merging**